libgraphics/include/pivot/graphics/VulkanException.hxx

```diff
@@ -1,20 +1,53 @@
 #pragma once
 
 #include <stdexcept>
-#include <vulkan/vulkan.hpp>
 
-namespace pivot::graphics
+namespace pivot
 {
+/// Base class for pivot exceptions
+class PivotException : public std::exception
```

Comment:
Please use `std::runtime_error` as your base. Implementing an exception type storing a string is really hard, as copying the exception should never throw in exception handling context, and the string constructor can throw.